### PR TITLE
Fix improper display of errors on create/edit refund views

### PR DESCRIPTION
**Pull request by Droniu**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [x] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes MERX-505, MERX-504

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/orders/views/OrderTransactionRefundEdit/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      errorLines.push({
+        code: err.code,
+        field: err.field,
+        lines: err.addLines,
+        message: err.message,
+      } as OrderTransactionRefundError);
+
+      setLinesErrors(errorLines);
+    });
+  }
```
Comment (poulch) [inline src/orders/views/OrderTransactionRefundEdit/handlers.ts:60]: question: seems like total duplication  of what we have here: `src/orders/views/OrderTransactionRefundCreate/handlers.ts`. Could we wrap this into a hook and share this code?

Comment (Droniu) [reply src/orders/views/OrderTransactionRefundEdit/handlers.ts:60]: There are slight differences, e.g.:
`submitData.orderGrantRefundCreate?.errors` vs
`submitData.orderGrantRefundUpdate?.errors`

or
`OrderGrantRefundUpdateErrorCode` vs `OrderGrantRefundCreateErrorCode`

behavior for completed mutation is different (order navigation vs clearing errors). I believe wrapping this into single abstracted hook will not make this any more readable and it's going to be much easier to modify mutation handling on specific view if we keep them separate.

File: src/orders/views/OrderTransactionRefundCreate/handlers.ts [hunk 2]
```diff
@@ -49,9 +49,14 @@ export const handleRefundCreateComplete = ({
...
 
-  if (errors.length > 1) {
+  if (errors.length > 0) {
     errors.forEach((err: OrderGrantRefundCreateErrorFragment) => {
-      if (err.code !== OrderGrantRefundCreateErrorCode.REQUIRED) {
+      if (
+        ![
+          OrderGrantRefundCreateErrorCode.REQUIRED,
+          OrderGrantRefundCreateErrorCode.AMOUNT_GREATER_THAN_AVAILABLE,
+        ].includes(err.code)
```
Comment (andrzejewsky) [inline src/orders/views/OrderTransactionRefundCreate/handlers.ts:58]: suggestion: function or at least variable

File: src/orders/views/OrderTransactionRefundEdit/handlers.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return;
+  }
+
+  if (errors.length > 0) {
+    errors.forEach((err: OrderGrantRefundUpdateErrorFragment) => {
+      if (
+        ![
+          OrderGrantRefundUpdateErrorCode.REQUIRED,
+          OrderGrantRefundUpdateErrorCode.AMOUNT_GREATER_THAN_AVAILABLE,
+        ].includes(err.code)
```
Comment (andrzejewsky) [inline src/orders/views/OrderTransactionRefundEdit/handlers.ts:43]: issue: you have repeated the same piece of code, why don't re-use it? it's rather simple and these errors will be handled mostly in the same way so there is no reason to break DRY

**Review by andrzejewsky — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**